examples/ui-demo/src/state/index.tsx

```diff
@@ -0,0 +1,170 @@
+    throw new Error("config context must be present in root");
+  }
+
+  return configContext;
+}
+
+function convertDemoConfigToUiConfig(config: Config): AlchemyAccountsUIConfig {
+  const sections: AuthType[][] = [[{ type: "email" }]];
+
+  if (config.auth.showPasskey) {
+    sections.push([{ type: "passkey" }]);
+  }
+
+  if (config.auth.showSocial && config.auth.addGoogleAuth) {
+    if (config.auth.showPasskey) {
+      sections
+        .at(-1)!
+        .push({ type: "social", authProviderId: "google", mode: "popup" });
+    } else {
+      sections.push([
+        { type: "social", authProviderId: "google", mode: "popup" },
+      ]);
+    }
+  }
+
+  if (config.auth.showExternalWallets) {
+    sections.push([
+      {
+        type: "external_wallets",
+        walletConnect: { projectId: "30e7ffaff99063e68cc9870c105d905b" },
+      },
+    ]);
+  }
+
+  return {
+    illustrationStyle: config.ui.illustrationStyle,
+    auth: {
+      sections,
+      addPasskeyOnSignup: config.auth.addPasskey && config.auth.showPasskey,
+      header: (
+        <AuthCardHeader
+          theme={config.ui.theme}
+          logoDark={config.ui.logoDark}
+          logoLight={config.ui.logoLight}
+        />
+      ),
+    },
+  };
+}
+
+export function ConfigContextProvider(
+  props: PropsWithChildren<{
+    initialConfig?: Config;
+    initialState?: AlchemyClientState;
+  }>
+) {
+  const storeRef = useRef<ReturnType<typeof createDemoStore>>();
+  if (!storeRef.current) {
+    storeRef.current = createDemoStore(props.initialConfig);
+  }
+
+  const { config, updateConfig: setConfig } = useStore(storeRef.current);
+  const [nftTransfered, setNFTTransfered] = useState(false);
+
+  // Sync Alchemy auth UI config
+  const uiConfig = useMemo(() => {
+    return convertDemoConfigToUiConfig(config);
+  }, [config]);
+
+  const localAlchemyConfig = useMemo(
+    () => ({
+      ...alchemyConfig,
+      ui: uiConfig,
+    }),
+    [uiConfig]
+  );
+
+  return (
+    <ConfigContext.Provider
+      value={{
+        config,
+        setConfig,
+        nftTransfered,
+        setNFTTransfered,
+      }}
+    >
+      <AlchemyAccountProvider
+        config={localAlchemyConfig}
+        queryClient={queryClient}
+        initialState={props.initialState}
+      >
+        {props.children}
+      </AlchemyAccountProvider>
+    </ConfigContext.Provider>
+  );
+}
+
+export function ConfigSync(props: PropsWithChildren) {
+  const { config } = useConfig();
+  const { updateConfig } = useUiConfig();
+
+  // Sync UI config changes
+  useEffect(() => {
+    const uiConfig = convertDemoConfigToUiConfig(config);
+    updateConfig(uiConfig);
+  }, [config, updateConfig]);
+
+  // Sync CSS variables
+  useEffect(() => {
+    const styles = generateStylesForRoot(config);
+    const classes = generateClassesForRoot(config);
+    const root = document.querySelector(":root") as HTMLElement;
+
+    styles.forEach((x) => root.style.setProperty(x[0], x[1]));
```

Comment:
Will there be any issues with styles not being removed if the entry was in the previous version of styles but not the current one?

Reply:
there might be yea, lemme do the same diffing logic like in the class list here

Reply:
ah yea it's not easy to do the diffing actually with the styles interface. But since the `generateStyles` returns the same properties for both the initial render and all updates change the values of those variables, I think we're ok here